cpp/memilio/compartments/simulation.h

```diff
@@ -140,11 +140,323 @@ class Simulation
         return m_integrator.get_dt();
     }
 
+protected:
+    /**
+     * @brief setup the simulation with an ODE solver
+     * @param[in] model: An instance of a compartmental model
+     * @param[in] t0 start time
+     * @param[in] dt initial step size of integration
+     */
+    template <class ODESystem>
+    Simulation(Model const& model, double t0, double dt, ODESystem& sys)
+        : m_integratorCore(
+              std::make_shared<mio::ControlledStepperWrapper<boost::numeric::odeint::runge_kutta_cash_karp54>>())
+        , m_model(std::make_unique<Model>(model))
+        , m_integrator(
+              [&model = *m_model, &sys](auto&& y, auto&& t, auto&& dydt) {
+                  sys.right_hand_side(model, y, t, dydt);
+              },
+              t0, sys.initial_values(model), dt, m_integratorCore)
+    {
+    }
+
 private:
     std::shared_ptr<IntegratorCore> m_integratorCore;
     std::unique_ptr<Model> m_model;
     OdeIntegrator m_integrator;
-}; // namespace mio
+};
+
+template <class M>
+class SimulationFlows : public Simulation<M>
+{
+public:
+    using Model = M;
+    using Base  = Simulation<M>;
+
+    /**
+     * @brief setup the simulation with an ODE solver
+     * @param[in] model: An instance of a compartmental model
+     * @param[in] t0 start time
+     * @param[in] dt initial step size of integration
+     */
+    SimulationFlows(Model const& model, double t0 = 0., double dt = 0.1)
+        : SimulationFlows(model, t0, dt, std::make_unique<ODESystem>(model.populations.numel()))
+    {
+    }
+
+    /**
+     * @brief get_result returns the values for all compartments within the
+     * simulated model for each time step.
+     * @return a TimeSeries to represent a numerical solution for the model. 
+     * For each simulated time step, the TimeSeries containts the population  
+     * size for each compartment. 
+     */
+    TimeSeries<ScalarType>& get_result()
+    {
+        // overwrite get_result from the base class, so that it also returns compartments (instead of flows)
+        compute_population_results();
+        return m_result;
+    }
```

Comment:
does this (only storing the flows and updating the result on demand) work for Graph/Migration simulations? Migration updates the result (whatever `get_result()` returns) at the last time point. Here, this update would be written to `SimulationFlows::m_result`, but as far as I can tell that isn't used during simulation.

Reply:
My solution is not particularly straightforward, so please have a look at the in-code documentation